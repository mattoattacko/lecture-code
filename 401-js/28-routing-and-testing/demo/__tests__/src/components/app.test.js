import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../../src/components/app.js';

describe('<App /> (Enzyme Test)', () => {

  it('is alive at application start', () => {
    let app = mount(<App />);
    expect(app.find('main #home').exists()).toBeTruthy();
  });

});

describe('<App /> (Snapshot Test)', () => {

  it('renders cleanly', () => {

    const component = renderer.create(
      <App/>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

  });

});

