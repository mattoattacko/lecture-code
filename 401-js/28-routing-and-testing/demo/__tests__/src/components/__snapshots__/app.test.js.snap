// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> (Snapshot Test) renders cleanly 1`] = `
Array [
  <header>
    <h1>
      Starter App
    </h1>
    <nav>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/items"
            onClick={[Function]}
          >
            List
          </a>
        </li>
        <li>
          <a
            href="/item"
            onClick={[Function]}
          >
            Item
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main />,
  <footer>
    <h3>
      Some cool stuff for the footer here
    </h3>
  </footer>,
]
`;
